### PR TITLE
Invite link

### PR DESCRIPTION
**Pull request by wburr10**

### CODE REVIEW

File: server/controllers/volunteerController.js [hunk 1]
```diff
@@ -359,10 +337,49 @@ let VolunteerController = function (volunteerRepository, shiftRepository, userRe
...
+            const emailClient = new EmailClient(emailClientTypes.NOREPLY);
+            emailClient.send(req.body.email,
+              "Invitation to City Harvest",
+              `Hello from Mobilise,\n
+          You have been invited to join City Harvest by ${req.user.firstName}.\n
+          Please click the following link to sign-up to Mobilise, the home of volunteering at City Harvest.\n\n
+          ${process.env.WEB_URL}/sign-up?token=${token}\n\n
+          This link will expire in 24 hours.`);
+            res.status(200).json({message: "Success! Invitation has been sent."});
+          })
```
Comment (sonjoonho) [inline server/controllers/volunteerController.js]: This promise does not get caught - is this intentional?

File: server/controllers/volunteerController.js [hunk 2]
```diff
@@ -359,10 +337,49 @@ let VolunteerController = function (volunteerRepository, shiftRepository, userRe
...
+
+  this.invite = function (req, res) {
+    if (!req.user.isAdmin) {
+      res.status(401).json({message: "Only admins can invite volunteers"});
+      return;
+    }
+    if (!req.user.email) {
+      res.status(400).json({message: "No email has been specified"});
+    }
+    volunteerRepository.getByEmail(req.body.email)
```
[the remark this answers is not in the document]
Comment (wburr10) [reply server/controllers/volunteerController.js]: Good spot, meant to be userRepository. I've changed it

File: server/controllers/volunteerController.js [hunk 3]
```diff
@@ -359,10 +337,49 @@ let VolunteerController = function (volunteerRepository, shiftRepository, userRe
...
+        const expires = moment().add(1, 'days').format();
+        return invitationTokenRepository.add(req.body.email, token, expires)
+          .then(() => {
+            const emailClient = new EmailClient(emailClientTypes.NOREPLY);
+            emailClient.send(req.body.email,
+              "Invitation to City Harvest",
+              `Hello from Mobilise,\n
+          You have been invited to join City Harvest by ${req.user.firstName}.\n
+          Please click the following link to sign-up to Mobilise, the home of volunteering at City Harvest.\n\n
+          ${process.env.WEB_URL}/sign-up?token=${token}\n\n
```
[the remark this answers is not in the document]
Comment (wburr10) [reply server/controllers/volunteerController.js]: Yes that is my bad. Have fixed now.

File: server/controllers/volunteerController.js (hunk 1 above)
Comment (wburr10) [reply server/controllers/volunteerController.js]: Ah wasn't aware it was a promise, will fix now

Comment (wburr10) [reply server/controllers/volunteerController.js]: Done

**Pull request merged**